### DISCUSSION TITLE
Identify and replace non-breaking spaces

### DISCUSSION THREAD

**Discussion by maxlouisr** [pulsar-edit/.github#254] [Community Package Help]
Hello everyone, it seems a recent mac os might have broken something in the way Haskell deals with non-breaking spaces (such as when you hold alt + space bar). How would go about telling Pulsar to look for these in a document and replace them all ? And also how would you reconfigure that keyboard shortcut to just do a normal space ? Thanks a lot

**Comment by Daeraxa** [comment]
This sounds rather odd... I don't think that shortcut is a Pulsar thing, is this not something done via macOS?

As for replacing them you should just be able to past an example of the "bad" space in the `Find in current buffer` box and replace them as any other character.

**Reply by maxlouisr** [reply]
It seems like it is indeed defined in macOs. I have tried everything i could find online to disable it but nothing worked. Replacing does the trick very well in the meanwhile !

**Comment by savetheclocktower** [comment]
In your `init.js` you can define a command for this.

```js
atom.commands.add('atom-text-editor', {
  'custom:insert-space': (event) => {
    let editor = atom.workspace.getActiveTextEditor();
    editor.insertText(' ');
  }
});
```

Then you can map alt-space to this command in your `keymap.cson`:

```coffeescript
'atom-text-editor':
  'alt-space': 'custom:insert-space'
```

Haven't tested this, so I might've made a silly mistake. Let me know if it doesn't work.

**Reply by maxlouisr** [reply]
Thanks a lot ! I'm not sure where these files are however- I could see several init.js in the pulsar application folder (in src, lib, npm-init...) Is it any one of these ? And i haven't yet found anything named keymap.cson. Thanks for your help

**Reply by savetheclocktower** [reply]
There are menu entries — one named **Init File…** and one named **Keymap…** — that will open these files for editing. Since you're on macOS, they're in the **Pulsar** application menu at the top of your screen.

They're stored in your home folder (`~/.pulsar`).

**Reply by maxlouisr** [reply]
Thanks so much, that worked ! Really cool

**Comment by PrincipaldelReydeReyes** [comment]
'atom-text-editor':
  'alt-space': 'custom:insert-space'
